src/ck-core/envelope.h

```diff
@@ -61,9 +61,9 @@ class PathHistoryEnvelope {
   void reset();
   void print() const;
   /// Write a description of the path into the beginning of the provided buffer. The buffer ought to be large enough.
-  void printHTMLToString(char* buf) const{
+  void printHTMLToString(char* buf, int len) const{
     buf[0] = '\0';
-    sprintf(buf+strlen(buf), "Path Time=%lf<br> Sender idx=%d", (double)totalTime, (int)sender_history_table_idx);
+    snprintf(buf+strlen(buf), len - strlen(buf), "Path Time=%lf<br> Sender idx=%d", (double)totalTime, (int)sender_history_table_idx);
```

Comment:
I think the original code has a bug: the `buf[0] = '\0';` will cause `strlen(buf)` to be zero, making that term superfluous